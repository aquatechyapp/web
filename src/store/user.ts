import { create } from 'zustand';
import { devtools } from 'zustand/middleware';

import { UserSubscription } from '@/ts/enums/enums';

import { User } from '../ts/interfaces/User';

type Store = {
  user: User;
  isFreePlan: boolean;
  reachedPoolLimit: boolean;
  shouldDisableNewPools: boolean;
};

type Actions = {
  setUser: (user: User) => void;
};

const poolsLimitBySubscription = {
  [UserSubscription.FREE]: 30,
  [UserSubscription.GROW]: 1000
};

export const useUserStore = create<Store & Actions>()(
  devtools((set) => ({
    user: {} as User,
    setUser: (user: User) => {
      set({
        user,
        isFreePlan: user.subscription === UserSubscription.FREE,
        reachedPoolLimit: user.poolsCount >= poolsLimitBySubscription[user.subscription],
        shouldDisableNewPools: user.poolsCount >= poolsLimitBySubscription[user.subscription]
      });
    },
    isFreePlan: false,
    reachedPoolLimit: false,
    shouldDisableNewPools: false
  }))
);
