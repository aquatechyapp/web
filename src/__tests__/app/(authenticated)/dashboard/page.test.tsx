import { render } from '@testing-library/react';
import { Mock } from 'vitest';

import DashboardPage from '@/app/(authenticated)/dashboard/page';
import useGetClients from '@/hooks/react-query/clients/getClients';
import { useUserStore } from '@/store/user';

vi.mock('@/store/user');
vi.mock('@/hooks/react-query/clients/getClients');

describe('DashboardPage', () => {
  vi.mocked(useUserStore).mockReturnValue({
    user: {
      id: '1',
      email: ''
    }
  });

  vi.mocked(useGetClients as Mock).mockReturnValue({
    data: [
      {
        id: '1',
        name: 'Client 1',
        pools: [
          {
            id: '1',
            city: 'City 1'
          }
        ]
      }
    ],
    isLoading: false
  });
  it('should render DashboardPage', () => {
    const wrapper = render(<DashboardPage />);
    expect(wrapper).toMatchSnapshot();
  });
});
