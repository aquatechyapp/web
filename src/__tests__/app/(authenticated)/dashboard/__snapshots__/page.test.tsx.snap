// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DashboardPage > should render DashboardPage 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h2
          class="text-gray-800 text-lg font-semibold mb-2"
        >
          September 2024
        </h2>
        <div
          class="flex w-full flex-wrap gap-6 text-nowrap"
        >
          <div
            class="flex flex-1 flex-col items-start gap-6"
          >
            <div
              class="inline-flex w-full flex-col items-start justify-start gap-4 rounded-lg border border-zinc-200 bg-white p-5"
            >
              <div
                class="flex flex-col items-start justify-start gap-2 self-stretch"
              >
                <div
                  class="inline-flex items-start justify-start gap-1 self-stretch"
                >
                  <div
                    class="flex h-6 w-6 items-center justify-center gap-2"
                  >
                    <svg
                      color="#2770ED"
                      fill="currentColor"
                      height="25"
                      stroke="currentColor"
                      stroke-width="0"
                      style="color: rgb(39, 112, 237);"
                      viewBox="0 0 24 24"
                      width="25"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M12.0049 22.0027C6.48204 22.0027 2.00488 17.5256 2.00488 12.0027C2.00488 6.4799 6.48204 2.00275 12.0049 2.00275C17.5277 2.00275 22.0049 6.4799 22.0049 12.0027C22.0049 17.5256 17.5277 22.0027 12.0049 22.0027ZM8.50488 14.0027V16.0027H11.0049V18.0027H13.0049V16.0027H14.0049C15.3856 16.0027 16.5049 14.8835 16.5049 13.5027C16.5049 12.122 15.3856 11.0027 14.0049 11.0027H10.0049C9.72874 11.0027 9.50488 10.7789 9.50488 10.5027C9.50488 10.2266 9.72874 10.0027 10.0049 10.0027H15.5049V8.00275H13.0049V6.00275H11.0049V8.00275H10.0049C8.62417 8.00275 7.50488 9.12203 7.50488 10.5027C7.50488 11.8835 8.62417 13.0027 10.0049 13.0027H14.0049C14.281 13.0027 14.5049 13.2266 14.5049 13.5027C14.5049 13.7789 14.281 14.0027 14.0049 14.0027H8.50488Z"
                      />
                    </svg>
                  </div>
                  <div
                    class="shrink grow basis-0 text-base font-medium leading-normal tracking-tight text-zinc-600"
                  >
                    Income
                    <span
                      class="text-xs"
                    >
                       (as a company)
                    </span>
                  </div>
                </div>
                <div
                  class="self-stretch text-[32px] font-medium tracking-tight text-neutral-800"
                >
                  $
                  NaN
                </div>
                <div
                  class="flex h-2 flex-col items-start justify-start gap-1 self-stretch"
                >
                  <div
                    class="flex h-2 flex-col items-start justify-start gap-2 self-stretch rounded-[100px] bg-gray-100 pr-28"
                  >
                    <div
                      class="h-2 w-[90%] rounded-[100px] bg-gradient-to-r from-blue-600 to-sky-400"
                    />
                  </div>
                </div>
              </div>
            </div>
            <a
              class="flex shrink grow basis-0 items-center justify-start gap-2 self-stretch rounded-lg border border-zinc-200 bg-gradient-to-b p-5 from-blue-600 to-sky-400"
              href="/clients/new"
            >
              <svg
                color="white"
                fill="currentColor"
                height="25"
                stroke="currentColor"
                stroke-width="0"
                style="color: white;"
                viewBox="0 0 512 512"
                width="25"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M363 277h-86v86h-42v-86h-86v-42h86v-86h42v86h86v42z"
                />
                <path
                  d="M256 90c44.3 0 86 17.3 117.4 48.6C404.7 170 422 211.7 422 256s-17.3 86-48.6 117.4C342 404.7 300.3 422 256 422c-44.3 0-86-17.3-117.4-48.6C107.3 342 90 300.3 90 256c0-44.3 17.3-86 48.6-117.4C170 107.3 211.7 90 256 90m0-42C141.1 48 48 141.1 48 256s93.1 208 208 208 208-93.1 208-208S370.9 48 256 48z"
                />
              </svg>
              <div
                class="inline-flex shrink grow basis-0 flex-col items-start justify-start gap-2 self-stretch"
              >
                <div
                  class="self-stretch text-lg font-bold text-gray-50"
                >
                  Add client
                </div>
              </div>
            </a>
            <div
              class="inline-flex h-full w-full flex-col items-center justify-start gap-4 rounded-lg border border-gray-200 bg-gray-50 p-6"
            >
              <div
                class="inline-flex items-start justify-start gap-3 self-stretch"
              >
                <div
                  class="inline-flex shrink grow basis-0 flex-col items-start justify-start gap-0.5"
                >
                  <div
                    class="inline-flex items-baseline gap-3 self-stretch"
                  >
                    <div
                      class="text-xl font-semibold text-gray-800"
                    >
                      Pools by city
                    </div>
                    <div
                      class="text-sm font-medium text-gray-500"
                    >
                       (as a company)
                    </div>
                  </div>
                  <div
                    class="text-sm font-medium text-gray-500"
                  />
                </div>
              </div>
              <div
                class="flex h-80 flex-col items-start justify-start gap-4 self-stretch overflow-auto"
              >
                <div
                  class="inline-flex items-center justify-start gap-2 self-stretch border-b-[1px]"
                >
                  <div
                    class="ImgText flex h-10 shrink grow basis-0 items-center justify-center gap-2"
                  >
                    <div
                      class="Text inline-flex shrink grow basis-0 flex-col items-start justify-center gap-0.5"
                    >
                      <div
                        class="self-stretch text-sm font-medium text-gray-800"
                      >
                        City 1
                      </div>
                      <div
                        class="self-stretch text-xs font-normal leading-[18px] text-gray-500"
                      >
                        1 pools
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="flex flex-1 flex-col items-start gap-6"
          >
            <div
              class="inline-flex w-full flex-col items-start justify-start gap-4 rounded-lg border border-zinc-200 bg-white p-5"
            >
              <div
                class="flex flex-col items-start justify-start gap-2 self-stretch"
              >
                <div
                  class="inline-flex items-start justify-start gap-1 self-stretch"
                >
                  <div
                    class="flex h-6 w-6 items-center justify-center gap-2"
                  >
                    <svg
                      color="#45CB99"
                      fill="currentColor"
                      height="25"
                      stroke="currentColor"
                      stroke-width="0"
                      style="color: rgb(69, 203, 153);"
                      viewBox="0 0 576 512"
                      width="25"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M504.717 320H211.572l6.545 32h268.418c15.401 0 26.816 14.301 23.403 29.319l-5.517 24.276C523.112 414.668 536 433.828 536 456c0 31.202-25.519 56.444-56.824 55.994-29.823-.429-54.35-24.631-55.155-54.447-.44-16.287 6.085-31.049 16.803-41.548H231.176C241.553 426.165 248 440.326 248 456c0 31.813-26.528 57.431-58.67 55.938-28.54-1.325-51.751-24.385-53.251-52.917-1.158-22.034 10.436-41.455 28.051-51.586L93.883 64H24C10.745 64 0 53.255 0 40V24C0 10.745 10.745 0 24 0h102.529c11.401 0 21.228 8.021 23.513 19.19L159.208 64H551.99c15.401 0 26.816 14.301 23.403 29.319l-47.273 208C525.637 312.246 515.923 320 504.717 320zM403.029 192H360v-60c0-6.627-5.373-12-12-12h-24c-6.627 0-12 5.373-12 12v60h-43.029c-10.691 0-16.045 12.926-8.485 20.485l67.029 67.029c4.686 4.686 12.284 4.686 16.971 0l67.029-67.029c7.559-7.559 2.205-20.485-8.486-20.485z"
                      />
                    </svg>
                  </div>
                  <div
                    class="shrink grow basis-0 text-base font-medium leading-normal tracking-tight text-zinc-600"
                  >
                    Income
                    <span
                      class="text-xs"
                    >
                       (as a sub-contractor)
                    </span>
                  </div>
                </div>
                <div
                  class="self-stretch text-[32px] font-medium tracking-tight text-neutral-800"
                >
                  $
                  NaN
                </div>
                <div
                  class="flex h-2 flex-col items-start justify-start gap-1 self-stretch"
                >
                  <div
                    class="flex h-2 flex-col items-start justify-start gap-2 self-stretch rounded-[100px] bg-gray-100 pr-28"
                  >
                    <div
                      class="h-2 w-[90%] rounded-[100px] bg-gradient-to-r from-teal-400 to-yellow-400"
                    />
                  </div>
                </div>
              </div>
            </div>
            <a
              class="flex shrink grow basis-0 items-center justify-start gap-2 self-stretch rounded-lg border border-zinc-200 bg-gradient-to-b p-5 from-emerald-400 to-stone-400"
              href="/routes"
            >
              <svg
                color="white"
                fill="none"
                height="24"
                size="25"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  fill="currentColor"
                  opacity="0.9"
                >
                  <path
                    d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7ZM7 9c0-2.76 2.24-5 5-5s5 2.24 5 5c0 2.88-2.88 7.19-5 9.88C9.92 16.21 7 11.85 7 9Z"
                  />
                  <path
                    d="M12 11.5a2.5 2.5 0 1 0 0-5 2.5 2.5 0 0 0 0 5Z"
                  />
                </g>
              </svg>
              <div
                class="inline-flex shrink grow basis-0 flex-col items-start justify-start gap-2 self-stretch"
              >
                <div
                  class="self-stretch text-lg font-bold text-gray-50"
                >
                  Route dashboard
                </div>
              </div>
            </a>
            <div
              class="inline-flex h-full w-full flex-col items-center justify-start gap-4 rounded-lg border border-gray-200 bg-gray-50 p-6"
            >
              <div
                class="inline-flex items-start justify-start gap-3 self-stretch"
              >
                <div
                  class="inline-flex shrink grow basis-0 flex-col items-start justify-start gap-0.5"
                >
                  <div
                    class="inline-flex items-baseline gap-3 self-stretch"
                  >
                    <div
                      class="text-xl font-semibold text-gray-800"
                    >
                      Pools by city
                    </div>
                    <div
                      class="text-sm font-medium text-gray-500"
                    >
                       (as a subcontractor)
                    </div>
                  </div>
                  <div
                    class="text-sm font-medium text-gray-500"
                  />
                </div>
              </div>
              <div
                class="flex h-80 flex-col items-start justify-start gap-4 self-stretch overflow-auto"
              >
                <div>
                  No pools found
                </div>
              </div>
            </div>
          </div>
          <div
            class="flex flex-1 flex-col items-start gap-6"
          >
            <div
              class="inline-flex w-full flex-col items-start justify-start gap-4 rounded-lg border border-zinc-200 bg-white p-5"
            >
              <div
                class="flex flex-col items-start justify-start gap-2 self-stretch"
              >
                <div
                  class="inline-flex items-start justify-start gap-1 self-stretch"
                >
                  <div
                    class="flex h-6 w-6 items-center justify-center gap-2"
                  >
                    <svg
                      color="#7826CF"
                      fill="currentColor"
                      height="25"
                      stroke="currentColor"
                      stroke-width="0"
                      style="color: rgb(120, 38, 207);"
                      viewBox="0 0 512 512"
                      width="25"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M304 256c52.805 0 96-43.201 96-96s-43.195-96-96-96-96 43.201-96 96 43.195 96 96 96zm0 48c-63.598 0-192 32.402-192 96v48h384v-48c0-63.598-128.402-96-192-96zM112 224v-64H80v64H16v32h64v64h32v-64h64v-32h-64z"
                      />
                    </svg>
                  </div>
                  <div
                    class="shrink grow basis-0 text-base font-medium leading-normal tracking-tight text-zinc-600"
                  >
                    Clients
                    <span
                      class="text-xs"
                    />
                  </div>
                </div>
                <div
                  class="self-stretch text-[32px] font-medium tracking-tight text-neutral-800"
                >
                  1
                </div>
                <div
                  class="flex h-2 flex-col items-start justify-start gap-1 self-stretch"
                >
                  <div
                    class="flex h-2 flex-col items-start justify-start gap-2 self-stretch rounded-[100px] bg-gray-100 pr-28"
                  >
                    <div
                      class="h-2 w-[90%] rounded-[100px] bg-gradient-to-r from-purple-700 to-blue-600"
                    />
                  </div>
                </div>
              </div>
            </div>
            <a
              class="flex shrink grow basis-0 items-center justify-start gap-2 self-stretch rounded-lg border border-zinc-200 bg-gradient-to-b p-5 from-red-500 to-orange-300"
              href="/team"
            >
              <svg
                color="white"
                fill="none"
                height="24"
                size="25"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9 13.75c-2.34 0-7 1.17-7 3.5V19h14v-1.75c0-2.33-4.66-3.5-7-3.5ZM4.34 17c.84-.58 2.87-1.25 4.66-1.25s3.82.67 4.66 1.25H4.34ZM9 12c1.93 0 3.5-1.57 3.5-3.5S10.93 5 9 5 5.5 6.57 5.5 8.5 7.07 12 9 12Zm0-5c.83 0 1.5.67 1.5 1.5S9.83 10 9 10s-1.5-.67-1.5-1.5S8.17 7 9 7Zm7.04 6.81c1.16.84 1.96 1.96 1.96 3.44V19h4v-1.75c0-2.02-3.5-3.17-5.96-3.44ZM15 12c1.93 0 3.5-1.57 3.5-3.5S16.93 5 15 5c-.54 0-1.04.13-1.5.35.63.89 1 1.98 1 3.15s-.37 2.26-1 3.15c.46.22.96.35 1.5.35Z"
                  fill="currentColor"
                  opacity="0.9"
                />
              </svg>
              <div
                class="inline-flex shrink grow basis-0 flex-col items-start justify-start gap-2 self-stretch"
              >
                <div
                  class="self-stretch text-lg font-bold text-gray-50"
                >
                  My team
                </div>
              </div>
            </a>
            <div
              class="inline-flex h-full w-full flex-col items-center justify-start gap-4 rounded-lg border border-gray-200 bg-gray-50 p-6"
            >
              <div
                class="inline-flex items-start justify-start gap-3 self-stretch"
              >
                <div
                  class="inline-flex shrink grow basis-0 flex-col items-start justify-start gap-0.5"
                >
                  <div
                    class="inline-flex items-baseline gap-3 self-stretch"
                  >
                    <div
                      class="text-xl font-semibold text-gray-800"
                    >
                      My Team
                    </div>
                    <div
                      class="text-sm font-medium text-gray-500"
                    />
                  </div>
                  <div
                    class="text-sm font-medium text-gray-500"
                  />
                </div>
              </div>
              <div
                class="flex h-80 flex-col items-start justify-start gap-4 self-stretch overflow-auto"
              >
                <div>
                  No team found
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h2
        class="text-gray-800 text-lg font-semibold mb-2"
      >
        September 2024
      </h2>
      <div
        class="flex w-full flex-wrap gap-6 text-nowrap"
      >
        <div
          class="flex flex-1 flex-col items-start gap-6"
        >
          <div
            class="inline-flex w-full flex-col items-start justify-start gap-4 rounded-lg border border-zinc-200 bg-white p-5"
          >
            <div
              class="flex flex-col items-start justify-start gap-2 self-stretch"
            >
              <div
                class="inline-flex items-start justify-start gap-1 self-stretch"
              >
                <div
                  class="flex h-6 w-6 items-center justify-center gap-2"
                >
                  <svg
                    color="#2770ED"
                    fill="currentColor"
                    height="25"
                    stroke="currentColor"
                    stroke-width="0"
                    style="color: rgb(39, 112, 237);"
                    viewBox="0 0 24 24"
                    width="25"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12.0049 22.0027C6.48204 22.0027 2.00488 17.5256 2.00488 12.0027C2.00488 6.4799 6.48204 2.00275 12.0049 2.00275C17.5277 2.00275 22.0049 6.4799 22.0049 12.0027C22.0049 17.5256 17.5277 22.0027 12.0049 22.0027ZM8.50488 14.0027V16.0027H11.0049V18.0027H13.0049V16.0027H14.0049C15.3856 16.0027 16.5049 14.8835 16.5049 13.5027C16.5049 12.122 15.3856 11.0027 14.0049 11.0027H10.0049C9.72874 11.0027 9.50488 10.7789 9.50488 10.5027C9.50488 10.2266 9.72874 10.0027 10.0049 10.0027H15.5049V8.00275H13.0049V6.00275H11.0049V8.00275H10.0049C8.62417 8.00275 7.50488 9.12203 7.50488 10.5027C7.50488 11.8835 8.62417 13.0027 10.0049 13.0027H14.0049C14.281 13.0027 14.5049 13.2266 14.5049 13.5027C14.5049 13.7789 14.281 14.0027 14.0049 14.0027H8.50488Z"
                    />
                  </svg>
                </div>
                <div
                  class="shrink grow basis-0 text-base font-medium leading-normal tracking-tight text-zinc-600"
                >
                  Income
                  <span
                    class="text-xs"
                  >
                     (as a company)
                  </span>
                </div>
              </div>
              <div
                class="self-stretch text-[32px] font-medium tracking-tight text-neutral-800"
              >
                $
                NaN
              </div>
              <div
                class="flex h-2 flex-col items-start justify-start gap-1 self-stretch"
              >
                <div
                  class="flex h-2 flex-col items-start justify-start gap-2 self-stretch rounded-[100px] bg-gray-100 pr-28"
                >
                  <div
                    class="h-2 w-[90%] rounded-[100px] bg-gradient-to-r from-blue-600 to-sky-400"
                  />
                </div>
              </div>
            </div>
          </div>
          <a
            class="flex shrink grow basis-0 items-center justify-start gap-2 self-stretch rounded-lg border border-zinc-200 bg-gradient-to-b p-5 from-blue-600 to-sky-400"
            href="/clients/new"
          >
            <svg
              color="white"
              fill="currentColor"
              height="25"
              stroke="currentColor"
              stroke-width="0"
              style="color: white;"
              viewBox="0 0 512 512"
              width="25"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M363 277h-86v86h-42v-86h-86v-42h86v-86h42v86h86v42z"
              />
              <path
                d="M256 90c44.3 0 86 17.3 117.4 48.6C404.7 170 422 211.7 422 256s-17.3 86-48.6 117.4C342 404.7 300.3 422 256 422c-44.3 0-86-17.3-117.4-48.6C107.3 342 90 300.3 90 256c0-44.3 17.3-86 48.6-117.4C170 107.3 211.7 90 256 90m0-42C141.1 48 48 141.1 48 256s93.1 208 208 208 208-93.1 208-208S370.9 48 256 48z"
              />
            </svg>
            <div
              class="inline-flex shrink grow basis-0 flex-col items-start justify-start gap-2 self-stretch"
            >
              <div
                class="self-stretch text-lg font-bold text-gray-50"
              >
                Add client
              </div>
            </div>
          </a>
          <div
            class="inline-flex h-full w-full flex-col items-center justify-start gap-4 rounded-lg border border-gray-200 bg-gray-50 p-6"
          >
            <div
              class="inline-flex items-start justify-start gap-3 self-stretch"
            >
              <div
                class="inline-flex shrink grow basis-0 flex-col items-start justify-start gap-0.5"
              >
                <div
                  class="inline-flex items-baseline gap-3 self-stretch"
                >
                  <div
                    class="text-xl font-semibold text-gray-800"
                  >
                    Pools by city
                  </div>
                  <div
                    class="text-sm font-medium text-gray-500"
                  >
                     (as a company)
                  </div>
                </div>
                <div
                  class="text-sm font-medium text-gray-500"
                />
              </div>
            </div>
            <div
              class="flex h-80 flex-col items-start justify-start gap-4 self-stretch overflow-auto"
            >
              <div
                class="inline-flex items-center justify-start gap-2 self-stretch border-b-[1px]"
              >
                <div
                  class="ImgText flex h-10 shrink grow basis-0 items-center justify-center gap-2"
                >
                  <div
                    class="Text inline-flex shrink grow basis-0 flex-col items-start justify-center gap-0.5"
                  >
                    <div
                      class="self-stretch text-sm font-medium text-gray-800"
                    >
                      City 1
                    </div>
                    <div
                      class="self-stretch text-xs font-normal leading-[18px] text-gray-500"
                    >
                      1 pools
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="flex flex-1 flex-col items-start gap-6"
        >
          <div
            class="inline-flex w-full flex-col items-start justify-start gap-4 rounded-lg border border-zinc-200 bg-white p-5"
          >
            <div
              class="flex flex-col items-start justify-start gap-2 self-stretch"
            >
              <div
                class="inline-flex items-start justify-start gap-1 self-stretch"
              >
                <div
                  class="flex h-6 w-6 items-center justify-center gap-2"
                >
                  <svg
                    color="#45CB99"
                    fill="currentColor"
                    height="25"
                    stroke="currentColor"
                    stroke-width="0"
                    style="color: rgb(69, 203, 153);"
                    viewBox="0 0 576 512"
                    width="25"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M504.717 320H211.572l6.545 32h268.418c15.401 0 26.816 14.301 23.403 29.319l-5.517 24.276C523.112 414.668 536 433.828 536 456c0 31.202-25.519 56.444-56.824 55.994-29.823-.429-54.35-24.631-55.155-54.447-.44-16.287 6.085-31.049 16.803-41.548H231.176C241.553 426.165 248 440.326 248 456c0 31.813-26.528 57.431-58.67 55.938-28.54-1.325-51.751-24.385-53.251-52.917-1.158-22.034 10.436-41.455 28.051-51.586L93.883 64H24C10.745 64 0 53.255 0 40V24C0 10.745 10.745 0 24 0h102.529c11.401 0 21.228 8.021 23.513 19.19L159.208 64H551.99c15.401 0 26.816 14.301 23.403 29.319l-47.273 208C525.637 312.246 515.923 320 504.717 320zM403.029 192H360v-60c0-6.627-5.373-12-12-12h-24c-6.627 0-12 5.373-12 12v60h-43.029c-10.691 0-16.045 12.926-8.485 20.485l67.029 67.029c4.686 4.686 12.284 4.686 16.971 0l67.029-67.029c7.559-7.559 2.205-20.485-8.486-20.485z"
                    />
                  </svg>
                </div>
                <div
                  class="shrink grow basis-0 text-base font-medium leading-normal tracking-tight text-zinc-600"
                >
                  Income
                  <span
                    class="text-xs"
                  >
                     (as a sub-contractor)
                  </span>
                </div>
              </div>
              <div
                class="self-stretch text-[32px] font-medium tracking-tight text-neutral-800"
              >
                $
                NaN
              </div>
              <div
                class="flex h-2 flex-col items-start justify-start gap-1 self-stretch"
              >
                <div
                  class="flex h-2 flex-col items-start justify-start gap-2 self-stretch rounded-[100px] bg-gray-100 pr-28"
                >
                  <div
                    class="h-2 w-[90%] rounded-[100px] bg-gradient-to-r from-teal-400 to-yellow-400"
                  />
                </div>
              </div>
            </div>
          </div>
          <a
            class="flex shrink grow basis-0 items-center justify-start gap-2 self-stretch rounded-lg border border-zinc-200 bg-gradient-to-b p-5 from-emerald-400 to-stone-400"
            href="/routes"
          >
            <svg
              color="white"
              fill="none"
              height="24"
              size="25"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="currentColor"
                opacity="0.9"
              >
                <path
                  d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7ZM7 9c0-2.76 2.24-5 5-5s5 2.24 5 5c0 2.88-2.88 7.19-5 9.88C9.92 16.21 7 11.85 7 9Z"
                />
                <path
                  d="M12 11.5a2.5 2.5 0 1 0 0-5 2.5 2.5 0 0 0 0 5Z"
                />
              </g>
            </svg>
            <div
              class="inline-flex shrink grow basis-0 flex-col items-start justify-start gap-2 self-stretch"
            >
              <div
                class="self-stretch text-lg font-bold text-gray-50"
              >
                Route dashboard
              </div>
            </div>
          </a>
          <div
            class="inline-flex h-full w-full flex-col items-center justify-start gap-4 rounded-lg border border-gray-200 bg-gray-50 p-6"
          >
            <div
              class="inline-flex items-start justify-start gap-3 self-stretch"
            >
              <div
                class="inline-flex shrink grow basis-0 flex-col items-start justify-start gap-0.5"
              >
                <div
                  class="inline-flex items-baseline gap-3 self-stretch"
                >
                  <div
                    class="text-xl font-semibold text-gray-800"
                  >
                    Pools by city
                  </div>
                  <div
                    class="text-sm font-medium text-gray-500"
                  >
                     (as a subcontractor)
                  </div>
                </div>
                <div
                  class="text-sm font-medium text-gray-500"
                />
              </div>
            </div>
            <div
              class="flex h-80 flex-col items-start justify-start gap-4 self-stretch overflow-auto"
            >
              <div>
                No pools found
              </div>
            </div>
          </div>
        </div>
        <div
          class="flex flex-1 flex-col items-start gap-6"
        >
          <div
            class="inline-flex w-full flex-col items-start justify-start gap-4 rounded-lg border border-zinc-200 bg-white p-5"
          >
            <div
              class="flex flex-col items-start justify-start gap-2 self-stretch"
            >
              <div
                class="inline-flex items-start justify-start gap-1 self-stretch"
              >
                <div
                  class="flex h-6 w-6 items-center justify-center gap-2"
                >
                  <svg
                    color="#7826CF"
                    fill="currentColor"
                    height="25"
                    stroke="currentColor"
                    stroke-width="0"
                    style="color: rgb(120, 38, 207);"
                    viewBox="0 0 512 512"
                    width="25"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M304 256c52.805 0 96-43.201 96-96s-43.195-96-96-96-96 43.201-96 96 43.195 96 96 96zm0 48c-63.598 0-192 32.402-192 96v48h384v-48c0-63.598-128.402-96-192-96zM112 224v-64H80v64H16v32h64v64h32v-64h64v-32h-64z"
                    />
                  </svg>
                </div>
                <div
                  class="shrink grow basis-0 text-base font-medium leading-normal tracking-tight text-zinc-600"
                >
                  Clients
                  <span
                    class="text-xs"
                  />
                </div>
              </div>
              <div
                class="self-stretch text-[32px] font-medium tracking-tight text-neutral-800"
              >
                1
              </div>
              <div
                class="flex h-2 flex-col items-start justify-start gap-1 self-stretch"
              >
                <div
                  class="flex h-2 flex-col items-start justify-start gap-2 self-stretch rounded-[100px] bg-gray-100 pr-28"
                >
                  <div
                    class="h-2 w-[90%] rounded-[100px] bg-gradient-to-r from-purple-700 to-blue-600"
                  />
                </div>
              </div>
            </div>
          </div>
          <a
            class="flex shrink grow basis-0 items-center justify-start gap-2 self-stretch rounded-lg border border-zinc-200 bg-gradient-to-b p-5 from-red-500 to-orange-300"
            href="/team"
          >
            <svg
              color="white"
              fill="none"
              height="24"
              size="25"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 13.75c-2.34 0-7 1.17-7 3.5V19h14v-1.75c0-2.33-4.66-3.5-7-3.5ZM4.34 17c.84-.58 2.87-1.25 4.66-1.25s3.82.67 4.66 1.25H4.34ZM9 12c1.93 0 3.5-1.57 3.5-3.5S10.93 5 9 5 5.5 6.57 5.5 8.5 7.07 12 9 12Zm0-5c.83 0 1.5.67 1.5 1.5S9.83 10 9 10s-1.5-.67-1.5-1.5S8.17 7 9 7Zm7.04 6.81c1.16.84 1.96 1.96 1.96 3.44V19h4v-1.75c0-2.02-3.5-3.17-5.96-3.44ZM15 12c1.93 0 3.5-1.57 3.5-3.5S16.93 5 15 5c-.54 0-1.04.13-1.5.35.63.89 1 1.98 1 3.15s-.37 2.26-1 3.15c.46.22.96.35 1.5.35Z"
                fill="currentColor"
                opacity="0.9"
              />
            </svg>
            <div
              class="inline-flex shrink grow basis-0 flex-col items-start justify-start gap-2 self-stretch"
            >
              <div
                class="self-stretch text-lg font-bold text-gray-50"
              >
                My team
              </div>
            </div>
          </a>
          <div
            class="inline-flex h-full w-full flex-col items-center justify-start gap-4 rounded-lg border border-gray-200 bg-gray-50 p-6"
          >
            <div
              class="inline-flex items-start justify-start gap-3 self-stretch"
            >
              <div
                class="inline-flex shrink grow basis-0 flex-col items-start justify-start gap-0.5"
              >
                <div
                  class="inline-flex items-baseline gap-3 self-stretch"
                >
                  <div
                    class="text-xl font-semibold text-gray-800"
                  >
                    My Team
                  </div>
                  <div
                    class="text-sm font-medium text-gray-500"
                  />
                </div>
                <div
                  class="text-sm font-medium text-gray-500"
                />
              </div>
            </div>
            <div
              class="flex h-80 flex-col items-start justify-start gap-4 self-stretch overflow-auto"
            >
              <div>
                No team found
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
